src/array_api_stubs/_draft/creation_functions.py

```diff
@@ -214,19 +215,36 @@ def eye(
     """
 
 
-def from_dlpack(x: object, /) -> array:
+def from_dlpack(
+    x: object, /, *,
+    device: Optional[device] = None,
+    copy: Optional[bool] = None
+) -> Union[array, Any]:
     """
     Returns a new array containing the data from another (array) object with a ``__dlpack__`` method.
 
     Parameters
     ----------
     x: object
         input (array) object.
+    device: Optional[device]
+        device on which to place the created array. If ``device`` is ``None`` and ``x`` supports DLPack, the output array device must be inferred from ``x.device``. Default: ``None``.
+
+        The v2023.12 standard only mandates that a compliant library must offer a way for ``from_dlpack`` to create an array on CPU (using
+        a library-specifc way to represent the CPU device (``kDLCPU`` in DLPack) e.g. a ``"CPU"`` string or a ``Device("CPU")`` object).
+        If the array library does not support the CPU device and needs to outsource to another (compliant) array library, it may do so
+        with a clear user documentation and/or run-time warning. If a copy must be made to enable this, and ``copy`` is set to ``False``,
+        the function must raise ``ValueError``.
+
+        Other kinds of devices will be considered for standardization in a future version of this API standard.
+    copy: Optional[bool]
+        boolean indicating whether or not to copy the input. If ``True``, the function must always copy. If ``False``, the function must never copy and must raise a ``BufferError`` in case a copy is deemed necessary (e.g. the producer disallows views). If ``None``, the function must reuse the existing memory buffer if possible and copy otherwise. Default: ``None``.
 
     Returns
     -------
-    out: array
-        an array containing the data in `x`.
+    out: Union[array, Any]
+        an array containing the data in ``x``. In the case that the compliant library does not support the given ``device`` out of box
```

Comment:
The "In the case ..." sentence is out of place here I think, and should be deleted. Returning a device from another library as an intermediate object from `__dlpack__` can be fine, but `from_dlpack` should only return its own array type. If it does not understand the input to `device=`, it should raise an appropriate error like `ValueError` or `TypeError` (doesn't need to be specified here).

Reply:
I am sorry, but I thought I [highlighted this issue and raised a discussion above](https://github.com/data-apis/array-api/pull/741#discussion_r1483777111), and both of you agreed it is not a problem? Let us continue in that thread, and focus on the first snippet there. 

> but `from_dlpack` should only return its own array type. If it does not understand the input to `device=`, it should raise an appropriate error like `ValueError` or `TypeError` (doesn't need to be specified here).

This is simply impossible when someone wants `cupy.from_dlpack(arr_cupy, device="cpu", copy=True)` to work, which defeats the purpose of this PR (have a way to get the array on host that is portable). In this case, the returned object has to be a `numpy.ndarray`.